assets/blocks/learner-courses-block/learner-courses-settings.test.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * External dependencies
+ */
+import '@testing-library/jest-dom';
+import { render, fireEvent } from '@testing-library/react';
+
+/**
+ * Internal dependencies
+ */
+import LearnerCoursesSettings from './learner-courses-settings';
+
+jest.mock( '@wordpress/block-editor', () => ( {
+	...jest.requireActual( '@wordpress/block-editor' ),
+	InspectorControls: ( { children } ) => children,
+	BlockControls: ( { children } ) => children,
+} ) );
+
+describe( '<LearnerCoursesSettings />', () => {
+	it( 'Should render the settings with the defined values', () => {
+		const attributes = {
+			courseDescriptionEnabled: true,
+			featuredImageEnabled: false,
+			courseCategoryEnabled: true,
+			progressBarEnabled: false,
+			layoutView: 'grid',
+		};
+		const { queryByLabelText, container } = render(
+			<LearnerCoursesSettings
+				attributes={ attributes }
+				setAttributes={ () => {} }
+			/>
+		);
+
+		expect( queryByLabelText( 'Course description' ).checked ).toEqual(
+			attributes.courseDescriptionEnabled
+		);
+
+		expect( queryByLabelText( 'Featured image' ).checked ).toEqual(
+			attributes.featuredImageEnabled
+		);
+
+		expect( queryByLabelText( 'Course category' ).checked ).toEqual(
+			attributes.courseCategoryEnabled
+		);
+
+		expect( queryByLabelText( 'Progress bar' ).checked ).toEqual(
+			attributes.progressBarEnabled
+		);
+
+		const [ , listViewButton, gridViewButton ] = container.querySelectorAll(
```

Comment:
Not the best approach to test it. But to test it correctly, getting by the label text, or anything similar, we need to update the `@wordpress/components` dependency, and now it's having other implications because we're using the specific version that is installed. At some point, we should confirm if we don't need this dependency anymore and remove it from the Sensei assets.

Reply:
Refactored to use `data-testid` in the https://github.com/Automattic/sensei/pull/4126/